daemon/cmd/state.go

```diff
@@ -375,35 +350,59 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState, endpoi
 	}
 	endpointCleanupCompleted.Wait()
 
+	// Wait on *all* restored endpoints for initial envoy policy computation.
+	// Its safe to do so here because, we expect all endpoints to close this channel,
+	// even if they are eventually deleted and never undergo regeneration.
 	go func() {
 		for _, ep := range state.restored {
 			<-ep.InitialEnvoyPolicyComputed
 		}
 		close(d.endpointInitialPolicyComplete)
 	}()
 
-	go func() {
-		regenerated, total := 0, 0
-		if len(state.restored) > 0 {
-			for buildSuccess := range epRegenerated {
-				if buildSuccess {
-					regenerated++
-				}
-				total++
-				if total >= len(state.restored) {
-					break
-				}
-			}
+	// Trigger regeneration for relevant restored endopints in a separate goroutine.
+	go d.handleRestoredEndpointsRegeneration(endpointsToRegenerate, endpointsRegenerator)
+}
+
+// This method is called during daemon start to asynchronously handle regeneration for restored
```

Comment:
nit: Function comments should describe the function itself, not the caller. The caller can always change, and that often happens without updating the function comments. This can make the comments misleading and confusing, which is often worse than having no comments.

```suggestion
// Trigger asynchronous regeneration of restored endpoints.
```